### PR TITLE
Power Of Three

### PR DESCRIPTION
**Pull request by messi-yang**
take it a look.
Can this answer be better or your have other idea about this problem :dancers:  ?

### CODE REVIEW

**Comment by acearth** [comment]
Power_3 = [1, 3, 9, 27, 81, 243, 729, 2187, 6561, 19683, 59049, 177147,
           531441, 1594323, 4782969, 14348907, 43046721, 129140163, 387420489,
           1162261467, 3486784401, 10460353203, 31381059609, 94143178827, 282429536481,
           847288609443, 2541865828329, 7625597484987,
           22876792454961, 68630377364883, 205891132094649]
def is_power_of_three(n)
    Power_3.include?(n)
end

Could you help me improve my solution above ?  

Remind:
1.  Do not use too much multiply and division;
2. Avoid repeated computation.

File: power_of_three [hunk 1]
```diff
@@ -0,0 +1,17 @@
+def is_power_of_three(n)
+  n/=3 while n>1&&n%3==0
+  n==1?true:false
```
Comment (acearth) [inline power_of_three:3]: line 3 can be replaced by "n==1"

File: power_of_three [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+end
+
+a=1
+b=3
+c=9
+d=27
+e=77
+f=243
+p "in:#{a} , out:#{is_power_of_three(a)}"
```
Comment (acearth) [inline power_of_three:12]: you can test your method by 
is_power_of_three(3)
is_power_of_three(4)==false

.

Avoid to use unnecessary variables.

**Comment by messi-yang** [comment]
Power_3 = [1, 3, 9, 27, 81, 243, 729, 2187, 6561, 19683, 59049, 177147,
531441, 1594323, 4782969, 14348907, 43046721, 129140163, 387420489,
1162261467]

Maybe you just need that much integers in the array.
'Cause Integer has it's range from -2^31~2^31
I still don't have any idea to do without calculation

**Comment by acearth** [comment]
You can review my newest commitment

Notice,  you should consider number range is larger than Long's Range.

Use bsearch...

**Comment by messi-yang** [comment]
Cool , thank you !

**Comment by acearth** [comment]
Sometimes launch  issue is better for discuss.

**Comment by messi-yang** [comment]
Thanks a lot , I will use the correct way next time

**Pull request closed without merging**